website/scripts/prepare-sprint.js

```diff
@@ -28,14 +28,17 @@ module.exports = {
   fn: async function () {
 
     sails.log('Running custom shell script... (`sails run prepare-sprint`)');
+    
+    // Ensure milestone exists in both fleetdm/fleet and fleetdm/confidential
+    // TODO
```

Comment:
It stinks that, with milestones, unlike "Iteration" fields, you have to create milestones by hand and set the date manually.  And it has to happen in both fleetdm/fleet and fleetdm/confidential, since milestones live at the repo level, rather than the org level.  Also GitHub doesn't show the end date on milestones on hover like it does for "Release" (our custom field that uses GitHub's new "Iteration" field type).  

Nevertheless, I'm still coming around to milestones for associating issues in a kind of "release archive" after the fact.  Is that what you were imagining @Luke Heath?  (Provided the automation cleanly preserves the "Release" field)

FYI you can do GitHub searches using e.g. `milestone:4.13.0` and it works properly across repos as long as the milestone has the same name in both repos.  So it's possible to get a [single list across both repos](https://github.com/issues?q=archived%3Afalse+org%3Afleetdm+is%3Aissue+is%3Aopen+milestone%3A4.13.0+), e.g. for use in writing release notes.  (Issues are sometimes easier to parse and/or have more context than commits.  At Sails Co., we tried both and found the writing of release notes was easier to delegate to other team members if issues were used to write them, rather than commit messages ± grim analysis of the code changes)

fyi @noahtalerman @mike-j-thomas

Reply:
We use [changes files](https://fleetdm.com/docs/contributing/committing-changes#changes-files) for generating release notes currently. It seems to work pretty well judging by the quality of release notes @noahtalerman puts out, but I'm open to feedback on how we could improve that process.

Reply:
The changes files and release notes are great, but they don't provide an archive of exactly which issues were closed as part of each release. For example, this provides a [nice time capsule of 4.12.0](https://github.com/fleetdm/fleet/milestone/21?closed=1). If we ever need more context on a release, this provides it. 

@mikermcneil Yes, I'm imagining milestones are created and assigned as part of closing out each sprint. I don't know if it would be necessary to have the same type of archive in the `confidential` repo. Those issues tend to close at various times due to various things that are not released-related. But, if you wanted to know everything in the company happening at each release, you could create the release milestones in each repo. 

If we wanted to use the "Iteration" field type while the issue is open that'd be fine. We'd just create the milestone, assign the issues, and close them out as part of this close/prepare sprint script.

Reply:
Got it.  OK I will finish wiring up this script on or before Thursday May 5.

Reply:
@lukeheath is this need still relevant?  I can work with Eric and Mike T to get it scheduled.

Reply:
@mikermcneil I've found a manual process that only takes about 5 minutes to close everything and prepare for the next sprint. It's actually kinda helpful because it gives me a chance to remove milestone tags from tickets that weren't included in the release, which commonly happens. So, I'd say this isn't relevant or needed right now. Thanks for your work on it, though!

Reply:
Sounds good!